Snowflake.Data.Tests/IntegrationTests/SFDbCommandIT.cs

```diff
@@ -1040,5 +1413,287 @@ public void TestGetQueryId()
+                    {
+                        Thread.Sleep(1000);
+                    }
+
+                    // Assert
+                    Assert.AreEqual(QueryStatus.FAILED_WITH_ERROR, cmd.GetQueryStatus(queryId));
+
+                    // Act
+                    var thrown = Assert.Throws<SnowflakeDbException>(() => cmd.GetResultsFromQueryId(queryId));
+
+                    // Assert
+                    Assert.IsTrue(thrown.Message.Contains("'FAKE_TABLE' does not exist"));
+                }
+
+                conn.Close();
+            }
+        }
+
+        [Test]
+        public void TestGetStatusOfInvalidQueryId()
+        {
+            string fakeQueryId = "fakeQueryId";
+
+            using (SnowflakeDbConnection conn = new SnowflakeDbConnection())
+            {
+                conn.ConnectionString = ConnectionString;
+                conn.Open();
+
+                using (SnowflakeDbCommand cmd = (SnowflakeDbCommand)conn.CreateCommand())
+                {
+                    // Act
+                    var thrown = Assert.Throws<Exception>(() => cmd.GetQueryStatus(fakeQueryId));
+
+                    // Assert
+                    Assert.IsTrue(thrown.Message.Contains($"The given query id {fakeQueryId} is not valid uuid"));
+                }
+
+                conn.Close();
+            }
+        }
+
+        [Test]
+        public void TestGetResultsOfInvalidQueryId()
+        {
+            string fakeQueryId = "fakeQueryId";
+
+            using (SnowflakeDbConnection conn = new SnowflakeDbConnection())
+            {
+                conn.ConnectionString = ConnectionString;
+                conn.Open();
+
+                using (SnowflakeDbCommand cmd = (SnowflakeDbCommand)conn.CreateCommand())
+                {
+                    // Act
+                    var thrown = Assert.Throws<Exception>(() => cmd.GetResultsFromQueryId(fakeQueryId));
+
+                    // Assert
+                    Assert.IsTrue(thrown.Message.Contains($"The given query id {fakeQueryId} is not valid uuid"));
+                }
+
+                conn.Close();
+            }
+        }
+
+        [Test, NonParallelizable]
+        public void TestGetStatusOfUnknownQueryId()
+        {
+            string unknownQueryId = "ab123cde-1cba-789a-987f-1234a56b789c";
+
+            using (SnowflakeDbConnection conn = new SnowflakeDbConnection())
+            {
+                conn.ConnectionString = ConnectionString;
+                conn.Open();
+
+                using (SnowflakeDbCommand cmd = (SnowflakeDbCommand)conn.CreateCommand())
+                {
+                    // Act
+                    var queryStatus = cmd.GetQueryStatus(unknownQueryId);
+
+                    // Assert
+                    Assert.AreEqual(QueryStatus.NO_DATA, queryStatus);
+                }
+
+                conn.Close();
+            }
+        }
+
+        [Test]
+        public void TestGetResultsOfUnknownQueryId()
```

Comment:
when I run this test it is hanging

Reply:
Could you try again and check if it still happens? The test runs for ~3.5 minutes because it retries until the max data retry is reached

Reply:
3,5 min is a lot of time. Let's exclude the retry logic into a separate class e.g. QueryResultsAwaiter.
This class would have the default configuration (you have to move it):
```
        private const int AsyncNoDataMaxRetry = 24;
        private readonly int[] _asyncRetryPattern = { 1, 1, 2, 3, 4, 8, 10 };
```

but would have also an internal constructor allowing to provide another configuration.
Then you could write a test for some shorter waiting scenario.
And we would not test it for the original configuration, because 3,5 min is too much of time.

The class SnowflakeDbCommand would hold QueryResultsAwaiter instance and use it.

Reply:
Done. The PR will already be merged but please feel free to let me know if there's any follow up changes you'd like to see